### PR TITLE
Remove the "must have used before" requirement.

### PR DESCRIPTION
**Pull request by danmarg**
We assume report-only mode + the ability to revoke policies provides
sufficient safety. So we will remove any need to first successfully
apply/validate a policy before using it in 'enforce' mode.

### CODE REVIEW

File: mta-sts.md [hunk 1]
```diff
@@ -583,8 +503,13 @@ func tryMxAccordingTo(message, mx, policy) {
...
   }
   status := !(tryStartTls(mx, &connection) && certMatches(connection, mx)) 
-  if !status {
-    // Report error establishing TLS or validating cert.
+  status = true
+  if !tryStartTls(mx, &connection) {
+    status = false
+    reportError(E_NO_VALID_TLS)
+  } else if certMatches(connection, mx) {
+    status =f alse
```
Comment (prbinu) [inline mta-sts.md]: typo false

File: mta-sts.md [hunk 2]
```diff
@@ -261,7 +220,7 @@ When sending mail via a "smart host"--an intermediate SMTP relay rather than the
 message recipient's server--compliant senders MUST treat the smart host domain
 as the policy domain for the purposes of policy discovery and application.
 
-# Policy Validation
+#Policy Validation
```
Comment (prbinu) [inline mta-sts.md]: Format error - may need to revert this change.

Comment (prbinu) [inline mta-sts.md]: 233 A policy which has been successfully used to deliver mail according to these
 234 constraints is said to be a "validated policy".
This statement may not be required

File: mta-sts.md [hunk 3]
```diff
@@ -342,25 +299,16 @@ An example control flow for a compliant sender consists of the following steps:
 1. Check for a cached, non-expired policy. If none exists, attempt to fetch a
```
Comment (prbinu) [inline mta-sts.md]: > 1. Check for a cached, non-expired policy
The statement is not clear on whether the expiry is based on max-age in the json policy file or DNS.

**Comment by danmarg** [comment]
Done.
